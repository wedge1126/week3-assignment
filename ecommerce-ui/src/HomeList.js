import React from 'react';
import PropTypes from 'prop-types';

import HomeCard from './HomeCard';

export default class HomeList extends React.Component {
  render() {
      const cards = this.props.homes.map((home, i) => <HomeCard home={home} onAddToCart={this.props.onAddToCart} key={i} />);
      return <div>
        <h3>Available Homes</h3>
        <div className="flex-container">
            {cards}
        </div>
      </div>
  }
}

HomeList.propTypes = {
    homes: PropTypes.arrayOf(PropTypes.shape({
        title: PropTypes.string.isRequired,
        houseType: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
        location: PropTypes.shape({
            city: PropTypes.string.isRequired,
            country: PropTypes.string.isRequired
        }).isRequired,
        payment: PropTypes.shape({
            cost: PropTypes.number.isRequired,
            description: PropTypes.string.isRequired
        }).isRequired,
        host: PropTypes.shape({
            name: PropTypes.string.isRequired,
            isSuperhost: PropTypes.bool.isRequired
        }).isRequired,
        rating: PropTypes.shape({
            stars: PropTypes.number.isRequired,
            reviews: PropTypes.number.isRequired
        })
    })),
    onAddToCart: PropTypes.func.isRequired
}