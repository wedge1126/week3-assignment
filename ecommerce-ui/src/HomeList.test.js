import React from 'react';
import { shallow } from 'enzyme';

import HomeList from './HomeList';

import testData from './_data/testData.json';

describe('HomeList', () => {
    it('should render the test data correctly', () => {
        const selected = [];
        const onAddToCart = jest.fn();
        const component = shallow(
            <HomeList homes={testData} selectedHomes={selected} onAddToCart={onAddToCart} />);
        
        expect(component).toMatchSnapshot();
    });
});