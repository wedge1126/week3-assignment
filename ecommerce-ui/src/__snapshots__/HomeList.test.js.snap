// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeList should render the test data correctly 1`] = `
<div>
  <h3>
    Available Homes
  </h3>
  <div
    className="flex-container"
  >
    <HomeCard
      active={true}
      home={
        Object {
          "host": Object {
            "isSuperhost": true,
            "name": "Robert",
          },
          "houseType": "Apartment",
          "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
          "location": Object {
            "city": "New York",
            "country": "USA",
          },
          "payment": Object {
            "cost": 105,
            "description": "Free cancellation",
          },
          "rating": Object {
            "reviews": 133,
            "stars": 3,
          },
          "title": "Centrally-located Manhattan studio",
        }
      }
      key="0"
      onAddToCart={[MockFunction]}
    />
    <HomeCard
      active={true}
      home={
        Object {
          "host": Object {
            "isSuperhost": true,
            "name": "Jamie",
          },
          "houseType": "Apartment",
          "image": "https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80",
          "location": Object {
            "city": "Denver",
            "country": "USA",
          },
          "payment": Object {
            "cost": 130,
            "description": "",
          },
          "rating": Object {
            "reviews": 381,
            "stars": 4.5,
          },
          "title": "Denver apartment outside Union Station",
        }
      }
      key="1"
      onAddToCart={[MockFunction]}
    />
    <HomeCard
      active={true}
      home={
        Object {
          "host": Object {
            "isSuperhost": false,
            "name": "Olivia",
          },
          "houseType": "Entire House",
          "image": "https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80",
          "location": Object {
            "city": "Stockholm",
            "country": "Sweden",
          },
          "payment": Object {
            "cost": 230,
            "description": "Free cancellation",
          },
          "rating": Object {
            "reviews": 83,
            "stars": 5,
          },
          "title": "Quaint 3-bedroom home in Gamla stan",
        }
      }
      key="2"
      onAddToCart={[MockFunction]}
    />
    <HomeCard
      active={true}
      home={
        Object {
          "host": Object {
            "isSuperhost": true,
            "name": "Sheev",
          },
          "houseType": "Entire Space Station",
          "image": "http://www.islandbreath.org/2015Year/02/150205deathstar.jpg",
          "location": Object {
            "city": "Endor",
            "country": "Endor System",
          },
          "payment": Object {
            "cost": 3200000000,
            "description": "",
          },
          "rating": Object {
            "reviews": 501,
            "stars": 3.2,
          },
          "title": "Quaint 31,400 kmÂ³ home above lovely forest",
        }
      }
      key="3"
      onAddToCart={[MockFunction]}
    />
  </div>
</div>
`;
